subsys/bluetooth/controller/ll_sw/ull_adv_sync.c

```diff
@@ -558,23 +743,99 @@ uint8_t ull_adv_sync_pdu_set_clear(struct ll_adv_set *adv,
+ * @param[in]  data             Pointer to data to be added to header and
+ *                              extra_data. Content depends on the value of
+ *                              @p hdr_add_fields.
+ * @param[out] ter_idx          Index of new PDU.
+ *
+ * @Note
+ * @p data content depends on the flag provided by @p hdr_add_fields:
+ * - ULL_ADV_PDU_HDR_FIELD_CTE_INFO:
+ *   # @p data->field_data points to single byte with CTEInfo field
+ *   # @p data->extra_data points to memory where is struct lll_df_adv_cfg
+ *     for LLL.
+ * - ULL_ADV_PDU_HDR_FIELD_AD_DATA:
+ *   # @p data->field_data points to memory where first byte
+ *     is size of advertising data, following byte is a pointer to actual
+ *     advertising data.
+ *   # @p data->extra_data is NULL
+ * - ULL_ADV_PDU_HDR_FIELD_AUX_PTR: # @p data parameter is not used
+ *
+ * @return Zero in case of success, other value in case of failure.
+ */
+uint8_t ull_adv_sync_pdu_set_clear(struct lll_adv_sync *lll_sync,
+				   struct pdu_adv *ter_pdu_prev,
+				   struct pdu_adv *ter_pdu,
+				   uint16_t hdr_add_fields,
+				   uint16_t hdr_rem_fields,
+				   struct adv_pdu_field_data *data)
+{
+	int err;
+
+	err = adv_sync_hdr_set_clear(lll_sync, ter_pdu_prev, ter_pdu,
 				     hdr_add_fields, hdr_rem_fields,
 				     (data ? data->field_data : NULL));
-	if (err) {
-		return err;
-	}
+
+	return err;
+}
 
 #if defined(CONFIG_BT_CTLR_DF_ADV_CTE_TX)
-	if (extra_data) {
-		adv_sync_extra_data_set_clear(extra_data_prev, extra_data,
-					      hdr_add_fields, hdr_rem_fields,
-					      (data ? data->extra_data : NULL));
+/* @brief Set or clear fields in extended advertising header and store
+ *        extra_data if requested.
+ *
+ * @param[in]  extra_data_prev  Pointer to previous content of extra_data.
+ * @param[in]  hdr_add_fields   Flag with information which fields add.
+ * @param[in]  hdr_rem_fields   Flag with information which fields remove.
+ * @param[in]  data             Pointer to data to be stored in extra_data.
+ *                              Content depends on the data depends on
+ *                              @p hdr_add_fields.
+ *
+ * @Note
+ * @p data depends on the flag provided by @p hdr_add_fields.
+ * Information about content of value may be found in description of
+ * @ref ull_adv_sync_pdu_set_clear.
+ *
+ * @return Zero in case of success, other value in case of failure.
+ */
+void ull_adv_sync_extra_data_set_clear(void *extra_data_prev,
+				       void *extra_data_new,
+				       uint16_t hdr_add_fields,
+				       uint16_t hdr_rem_fields,
+				       void *data)
+{
+	/* Currently only CTE enable requires extra_data. Due to that fact
+	 * CTE additional data are just copied to extra_data memory.
+	 */
+	if (hdr_add_fields & ULL_ADV_PDU_HDR_FIELD_CTE_INFO) {
+		memcpy(extra_data_new, data, sizeof(struct lll_df_adv_cfg));
+	} else if ((hdr_rem_fields & ULL_ADV_PDU_HDR_FIELD_CTE_INFO) ||
```

Comment:
```suggestion
} else if (!(hdr_rem_fields & ULL_ADV_PDU_HDR_FIELD_CTE_INFO) ||
```
This is an error that is available from first implementation of the code.